spec/requests/admin/enrollments_spec.rb

```diff
@@ -0,0 +1,390 @@
+                                    end_time: enrollment.end_time + 90.minutes,
+                                    member_id: member.id
+        attributes.merge!(convert_datetime_in_params(attributes[:start_time], 'start_time'))
+        attributes.merge!(convert_datetime_in_params(attributes[:end_time], 'end_time'))
+        attributes
+      end
+
+      let(:assign_other_members_to_this_mission) do
+        create_list :enrollment,
+                    4,
+                    start_time: enrollment.start_time + 90.minutes,
+                    end_time: enrollment.end_time + 90.minutes,
+                    mission_id: mission.id,
+                    member_id: (create :member).id
+      end
+
+      it 'displays an error message' do
+        assign_other_members_to_this_mission
+
+        put_enrollment
+
+        expect(response.body).to include(I18n.t('activerecord.errors.models.enrollment.slot_unavailability'))
+      end
+    end
+
+    context 'when the related mission is regulated and the member has an
+             insufficient cash register mastery for the new time slot' do
+      let(:mission) { create :mission, genre: 'regulated', cash_register_proficiency_requirement: 'proficient' }
+      let(:enrollment) do
+        create :enrollment,
+               start_time: mission.start_date,
+               end_time: mission.start_date + 90.minutes,
+               mission_id: mission.id
+      end
+      let(:enrollment_params) do
+        attributes = attributes_for :enrollment,
+                                    start_time: enrollment.start_time + 90.minutes,
+                                    end_time: enrollment.end_time + 90.minutes,
+                                    member_id: member.id
+        attributes.merge!(convert_datetime_in_params(attributes[:start_time], 'start_time'))
+        attributes.merge!(convert_datetime_in_params(attributes[:end_time], 'end_time'))
+        attributes
+      end
+
+      let(:assign_other_members_to_this_mission) do
+        create_list :enrollment,
+                    3,
+                    start_time: mission.start_date,
+                    end_time: mission.due_date,
+                    mission_id: mission.id,
+                    member_id: (create :member).id
+      end
+
+      i18n_key = <<~KEY.strip
+        activerecord.errors.models.enrollment.insufficient_cash_register_proficiency
+      KEY
+
+      it 'displays an error message' do
+        assign_other_members_to_this_mission
+
+        put_enrollment
+
+        expect(CGI.unescapeHTML(response.body)).to include(I18n.t(i18n_key))
+      end
+    end
+  end
+
+  # this helper imitate the struture of date_params created by active admin datepicker
+  def convert_datetime_in_params(datetime, key)
+    {
+      "#{key}(1i)": datetime.year,
+      "#{key}(2i)": datetime.month,
+      "#{key}(3i)": datetime.day,
+      "#{key}(4i)": datetime.hour,
+      "#{key}(5i)": datetime.min
+    }
+  end
```

Comment:
I think I saw this same helper at least 3 times. You could maybe extract it in its own service object